tests/NoUnused/ExportsTest.elm

```diff
@@ -1549,3 +1545,101 @@ unused = 1
                           )
                         ]
         ]
+
+
+exposingAllTests : Test
+exposingAllTests =
+    describe "When exposing all"
+        [ test "works with a function" <|
```

Comment:
Thank you for adding tests :pray: 

Would you mind adding some explicit tests where a module exposes everything (if you've noticed such tests somewhere else, feel free to move them here) but nothing gets reported because all their exposed elements are used? I'm thinking one test for functions, one for type alias, etc. just like you did here.
And I think we should do so for both usages in the same file and usages in other files.